myna/core/workflow/launch_from_peregrine.py

```diff
@@ -34,108 +34,127 @@ def working_directory(path):
+        help="list of part names to simulate",
+    )
+    parser.add_argument(
+        "--layers",
+        default=None,
+        type=lambda s: [int(item) for item in s.strip()[1:-1].split(",")],
+        help="list of layers to simulate",
+    )
+    parser.add_argument(
+        "--workspace",
+        default=None,
+        type=str,
+        help="path to the desired workspace to use",
+    )
+    parser.add_argument(
+        "--mode",
+        default=None,
+        type=str,
+        help="simulation mode to use",
+    )
+
+    args = parser.parse_args()
+
+    # Parse the arguments passed from Peregrine
+    build_path = os.path.abspath(args.build)
+    layers = args.layers
+    exported_parts = args.parts
+    mode = args.mode
+    workspace = os.path.abspath(args.workspace)
+
+    # Check if build path includes Peregrine as the last directory
+    if any([build_path[-1] == sep for sep in [os.path.sep, os.path.altsep]]):
+        build_path = build_path[:-1]
+    if os.path.basename(build_path) == "Peregrine":
+        build_path = os.path.dirname(build_path)
+
+    # Create Myna directory if it doesn't already exist
+    myna_working_dir = os.path.join(build_path, "Myna")
+    os.makedirs(myna_working_dir, exist_ok=True)
+
+    # Write Peregrine input to log file
+    lines = []
+    lines.append(f"Peregrine inputs:\n")
+    lines.append(f"- {build_path=}\n")
+    lines.append(f"- {layers=}\n")
+    lines.append(f"- {exported_parts=}\n")
+    lines.append(f"- {mode=}\n")
+    lines.append(f"- {workspace=}\n")
+
+    # Get yyyy-mm-dd_hh-mm format for the current time and add to log file
+    now = datetime.datetime.now()
+    now_str_pretty = now.strftime("%Y-%m-%d %H:%M:%S")
+    now_str_id = now.strftime("%Y-%m-%d-%Hh-%Mm")
+    lines.append(f"\nSimulation started at {now_str_pretty}\n")
+
+    # Set input file paths
+    input_file = os.path.join(peregrine_launcher_path, f"input_{mode}.yaml")
+    input_file_configured = os.path.join(
+        myna_working_dir, f"input_{mode}_{now_str_id}.yaml"
+    )
+
+    # Read and update input dictionary
+    output_dir = os.path.basename(build_path).replace(" ", "_") + f"_{now_str_id}"
+    with open(input_file, "r") as f:
+        input_dict = yaml.safe_load(f)
+    input_dict["data"] = {}
+    input_dict["data"]["build"] = {}
+    input_dict["data"]["build"]["datatype"] = "Peregrine"
+    input_dict["data"]["build"]["name"] = output_dir
+    input_dict["data"]["build"]["path"] = build_path
+    input_dict["data"]["build"]["parts"] = {}
+    for part in exported_parts:
+        input_dict["data"]["build"]["parts"][part] = {"layers": layers}
+    input_dict["myna"] = {}
+    input_dict["myna"]["workspace"] = workspace
+
+    # Export updated input dictionary
+    with open(input_file_configured, "w") as f:
+        yaml.dump(input_dict, f, default_flow_style=False)
+
+    # Set working directory to run all myna scripts
+    with working_directory(myna_working_dir):
```

Comment:
Not sure I understand what this provides

Reply:
I find the context manager syntax to be more intuitive/readable than bookending this section with `os.chdir(myna_working_dir)` & `os.chdir(original_dir)`. To me, it clearer what should be done in the scope of the directory.

Alternatively, everything could use absolute paths, which would make the working directory irrelevant.

Reply:
Sounds good, I just haven't used it in that way before